api/src/test/java/org/openmrs/module/m2sysbiometrics/service/SearchServiceTest.java

```diff
@@ -0,0 +1,93 @@
+package org.openmrs.module.m2sysbiometrics.service;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.openmrs.module.m2sysbiometrics.bioplugin.LocalBioServerClient;
+import org.openmrs.module.m2sysbiometrics.bioplugin.NationalBioServerClient;
+import org.openmrs.module.m2sysbiometrics.exception.M2SysBiometricsException;
+import org.openmrs.module.m2sysbiometrics.model.M2SysCaptureResponse;
+import org.openmrs.module.m2sysbiometrics.service.impl.SearchServiceImpl;
+import org.openmrs.module.m2sysbiometrics.testdata.M2SysCaptureResponseMother;
+import org.openmrs.module.registrationcore.api.biometrics.model.BiometricMatch;
+
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SearchServiceTest {
+
+    private static final String EXISTING_SUBJECT_ID = "ExistingSubjectId";
+
+    private static final String EXISTING_RESULT_XML = "<Results><result score='72' value='" + EXISTING_SUBJECT_ID +
+            "'></Results>";
+
+    private static final String ERROR_RESULT_XML = "<Results><result score='0' value='TEMPLATE_FORMAT_ERROR'></Results>";
```

Comment:
Add test case for:

<Results><result score="0" value="-1" Instance="2"/></Results>

which is no results

Reply:
ok I will correct it